src/session_impl.cpp

```diff
@@ -6198,41 +6200,59 @@ namespace aux {
 
 	void session_impl::set_local_download_rate_limit(int bytes_per_second)
 	{
+#if TORRENT_USE_INVARIANT_CHECKS
```

Comment:
the invariant of a class is the valid states. the invariant is supposed to be maintained at all times, except while executing member functions. a (public) member function may break the invariant temporarily, to re-establish it before it returns. The invariant check must be run at the entry of a function to make sure nothing unexpected has happened to cause the object to have an invalid state, and at the end it's checked to make sure the function correctly re-established the invariant.